lib/ansible/module_utils/connection.py

```diff
@@ -142,6 +142,27 @@ def _exec_jsonrpc(self, name, *args, **kwargs):
 
         try:
             data = json.dumps(req)
+        except TypeError as exc:
+            data = req['params']
+
+            if isinstance(data, dict):
+                data = data.get('var_options', {})
+
+                for key, value in iteritems(data):
+                    try:
+                        dummy = json.dumps(value)
+                    except TypeError:
+                        raise ConnectionError(
+                            "Failed to encode some variables as JSON for communication with ansible-connection. "
+                            "Please open an issue and mention that the culprit is most likely '%s'" % key
+                        )
+
+            raise ConnectionError(
+                "Failed to encode some variables as JSON for communication with ansible-connection. "
+                "The original exception was: " % to_text(exc)
```

Comment:
You're missing `%s` in the string. `to_text(exc)` is rendered to `/dev/null` now :)